pkg/mesh/backend.go

```diff
@@ -54,8 +56,9 @@ const (
 
 // Node represents a node in the network.
 type Node struct {
-	Endpoint     *wireguard.Endpoint
-	Key          []byte
+	Endpoint     *net.UDPAddr
+	Addr         string // eg. dnsname:port
```

Comment:
Before this change, we only stored three values: IP, DNS, and Port. If we keep this as Addr, then we are duplicating data, which leads to ambiguity IMO about which is the "real" port. Can we instead make this DNS? Also, if it really needs to have both fields, I think it would be better to make it a struct so we don't always have to split the string. WDYT?
```suggestion
	DNS         string
```

Reply:
I think it it is awkward to use the net.UDPAddr to only store the port, this would mean we would always have to check if the IP is nil, to see if the endpoint is invalid instead of just `updAddr != nil`.

If we split the dns name and port, more code would be neccesary to resolve the endpoint because right now, we can simply use `net.UDPResolve` which wants a string with IP and port. I think we only split the string once, when we check if it is valid. And we would still do this because it comes form the `kilo.squat.ai/force-endpoint` annotation for nodes or the `v1alpha1` API. So right now, I don't see the benefit of using a new struct with a String method.

Or we continue using the old `wireguard.Enpoint` and translate this to net.UDPAddr at some point in the code before we use compare old and new config and set the config. This feels like going backwards to me, but maybe it makes sense.

Reply:
I worried that we will somehow end up with different values for the ports in each field and that there will be some hard-to-track bugs as a result. This smells a bit to me like a leaky abstraction. Also, it's still confusing to me that we call this field Addr, when it is really only supposed to represent an optional DNS name.
Instead of ResolveUPDAddr, we can do a DNS lookup, which that func does under the hood anyways. That's actually exactly what we already so today: https://github.com/squat/kilo/blob/main/pkg/mesh/mesh.go#L656